Tests/scripts/validate_files_structure.py

```diff
@@ -682,6 +703,222 @@ def str2bool(v):
         raise argparse.ArgumentTypeError('Boolean value expected.')
 
 
+def is_text_file(file_path):
+    file_extension = os.path.splitext(file_path)[1]
+    text_file_types = set(['.yml', '.py', '.json', '.md', '.txt', '.sh', '.ini', '.eml', '', '.csv'])
+    if file_extension in text_file_types:
+            return True
+    return False
+
+
+def get_diff_text_files(files_string):
+    accepted_file_statuses = ['M', 'A']
+    all_files = files_string.split('\n')
+    text_files_list = set([])
+    for file_name in all_files:
+        file_data = file_name.split()
+        if not file_data:
+            continue
+
+        file_status = file_data[0]
+        file_path = file_data[1]
+
+        if file_status.upper() in accepted_file_statuses and is_text_file(file_path) and SECRETS_WHITE_LIST_FILE not in file_path:
+            text_files_list.add(file_path)
+
+    return text_files_list
+
+
+def get_all_diff_text_files(branch_name, is_circle):
+
+    if is_circle:
+        branch_changed_files_string = run_git_command("git diff --name-status origin/master...{}".format(branch_name))
+        text_files_list = get_diff_text_files(branch_changed_files_string)
+
+    else:
+        local_changed_files_string = run_git_command("git diff --name-status --no-merges HEAD")
+        text_files_list = get_diff_text_files(local_changed_files_string)
+
+    return text_files_list
+
+
+def validate_committed_files(branch_name, is_circle):
+
+    # secrets validation starts
+    # Get all files that have been modified & added to repo
+    secrets_file_paths = get_all_diff_text_files(branch_name, is_circle)
+    secrets_found = search_potential_secrets(secrets_file_paths)
+
+    if secrets_found:
+        print_error('Secrets were found in the following files:\n')
+        for file in secrets_found:
+            print_error('\nFile Name: ' + file)
+            print_error(json.dumps(secrets_found[file], indent=4))
+        print_error('Remove or whitelist secrets in order to proceed, then re-commit\n')
+
+        sys.exit(1)
+    # secrets validation ends
+
+    modified_files, added_files = get_modified_and_added_files(branch_name, is_circle)
+
+    with open('./Tests/id_set.json', 'r') as id_set_file:
+        id_set = json.load(id_set_file)
+
+    script_set = id_set['scripts']
+    playbook_set = id_set['playbooks']
+    integration_set = id_set['integrations']
+    test_playbook_set = id_set['TestPlaybooks']
+
+    has_schema_problem = validate_modified_files(integration_set, modified_files,
+                                                 playbook_set, script_set, test_playbook_set, is_circle)
+
+    has_schema_problem = validate_added_files(added_files, integration_set, playbook_set,
+                                              script_set, test_playbook_set, is_circle) or has_schema_problem
+
+    if has_schema_problem:
+        sys.exit(1)
+
+
+def search_potential_secrets(secrets_file_paths):
```

Comment:
is it possible to split the content of this function to several functions?
It's a bit long and complicated for a person to dive in

Reply:
I already split it into 5 functions, I can keep going with splitting but the majority is already out.